### PR TITLE
Splash screen

### PR DESCRIPTION
**Pull request by johangv**

### CODE REVIEW

File: lib/features/splash/domain/use_cases/get_splash_use_case.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  final SplashRepository repository;
+  GetSplashUseCase({required this.repository});
+
+  Future<bool> getUser() async {
+    final result = await repository.getSplash();
+
+    if (result) {
+      return true;
+    } else {
+      return false;
```
Comment (oscarshaitan) [inline lib/features/splash/domain/use_cases/get_splash_use_case.dart]: just return the result

File: lib/features/splash/presentation/cubit/splash_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  void init() async {
+    final result = await _getSplashUseCase.getUser();
+
+    if (result) {
+      emit(SplashState.logedUser);
+    } else {
+      emit(SplashState.none);
+    }
+  }
+}
```
Comment (oscarshaitan) [inline lib/features/splash/presentation/cubit/splash_cubit.dart]: good cubit implementation providing the dependency

File: lib/features/splash/presentation/cubit/splash_state.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+class Loading extends UserState {
+  const Loading();
+}
+
+class Loaded extends UserState {
+  const Loaded(this.user);
+  final bool user;
+}
```
Comment (oscarshaitan) [inline lib/features/splash/presentation/cubit/splash_state.dart]: look for how to create a state with freezed is quite similar on what you have here but give us more tools to work

File: lib/features/splash/presentation/cubit/splash_state.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+part of 'splash_cubit.dart';
+
+abstract class UserState {
+  const UserState();
+}
```
Comment (oscarshaitan) [inline lib/features/splash/presentation/cubit/splash_state.dart]: why the file is splash state but the class ios user state?

**Review by oscarshaitan — COMMENTED** [review]
This is a nice PR
and a good example on how to do the proper CLEAN Arch...
but for this project lets skip the use cases and provide the repository.
the only reason is that 90% of our use case are going to do simply calling the repo

also lets implement get_it to not use the provider  to provide dependencies 
get_it will help us for other tasks like save a session state 

but in general goood job!

File: lib/features/splash/domain/use_cases/get_splash_use_case.dart (hunk 1 above)
Comment (johangv) [reply lib/features/splash/domain/use_cases/get_splash_use_case.dart]: Fixed

File: lib/features/splash/presentation/cubit/splash_state.dart (hunk 4 above)
Comment (johangv) [reply lib/features/splash/presentation/cubit/splash_state.dart]: Fixed

File: lib/features/splash/presentation/cubit/splash_state.dart (hunk 3 above)
Comment (johangv) [reply lib/features/splash/presentation/cubit/splash_state.dart]: Done

**Comment by johangv** [comment]
Hello Oscar, thanks for the suggestions.
I'm fixed the things that you mentioned in the comments. And, as you can see in the last commit I'm implemented freezed notation in the cubit states, I'm replace provider with get_it and delete the uses cases.
I await new comments, have a nice day!!

**Review by oscarshaitan — APPROVED** [review]

**Pull request merged**